app/assets/javascripts/darkswarm/services/credit_cards.js.coffee

```diff
@@ -11,5 +11,14 @@ Darkswarm.factory 'CreditCards', ($http, $filter, savedCreditCards, Messages)->
         othercard.is_default = false
       $http.put("/credit_cards/#{card.id}", is_default: true).then (data) ->
         Messages.success(t('js.default_card_updated'))
+        for customer in Customers.index()
+          customer.allow_charges = false
```

Comment:
If we wanted, we could have this refresh the Customers state from the server, probably using a `$watch` on the form? My Angular skills weren't quite up to figuring that out... :)

Reply:
the allow charges check boxes will need to be cleared here.
I think there's a design problem here, this is credit cards, it should handle credit cards only. I believe these two lines should be in Customers service, not here. Something like Customers.clear_all_allow_charges ?

Reply:
👍 6e7cc1163ca4bfd1e02a33d49303663059d0d726